src/domain/templates/components/TemplateSelectors/SubspaceTemplateSelector.tsx

```diff
@@ -1,26 +1,55 @@
 import { Box, Button, Chip, Skeleton } from '@mui/material';
-import { useField } from 'formik';
+import { useField, useFormikContext } from 'formik';
 import React, { FC, useMemo, useState } from 'react';
 import { useTranslation } from 'react-i18next';
 import { BlockSectionTitle, Caption } from '@/core/ui/typography';
 import ImportTemplatesDialog from '../Dialogs/ImportTemplateDialog/ImportTemplatesDialog';
 import SystemUpdateAltIcon from '@mui/icons-material/SystemUpdateAlt';
 import { LibraryIcon } from '@/domain/templates/LibraryIcon';
-import { SpaceLevel, TemplateDefaultType, TemplateType } from '@/core/apollo/generated/graphql-schema';
-import { useSpaceDefaultTemplatesQuery, useTemplateNameQuery } from '@/core/apollo/generated/apollo-hooks';
+import { SpaceLevel, TemplateDefaultType, TemplateType, VisualType } from '@/core/apollo/generated/graphql-schema';
+import {
+  useSpaceDefaultTemplatesQuery,
+  useTemplateNameQuery,
+  useTemplateContentLazyQuery,
+} from '@/core/apollo/generated/apollo-hooks';
 import { Identifiable } from '@/core/utils/Identifiable';
 import Gutters, { GuttersProps } from '@/core/ui/grid/Gutters';
 import useUrlResolver from '@/main/routing/urlResolver/useUrlResolver';
+import { getVisualByType } from '@/domain/common/visual/utils/visuals.utils';
+import { VisualModel } from '@/domain/common/visual/model/VisualModel';
+import ConfirmationDialog from '@/core/ui/dialogs/ConfirmationDialog';
+
+export type BasicVisualUrlModel = {
```

Comment:
I think we could make this a common thing. I'd move it do `domain/common/visual/utils`
And call it something like `EntityVisualUrls`
have there the four types:
```
export type EntityVisualUrls = {
  avatarUrl?: string;
  cardBannerUrl?: string;
  bannerUrl?: string;
  bannerWideUrl?: string;
}
```
and the function `getVisualUrls` moved together there